atomica/plotting.py

```diff
@@ -1606,36 +1613,53 @@ def _turn_off_border(ax) -> None:
     ax.yaxis.set_ticks_position('left')
 
 
-def plot_legend(entries: dict, plot_type='patch', fig=None):
+def plot_legend(entries: dict, plot_type='patch', fig=None, legendsettings=None):
     """
     Render a new legend
 
     :param entries: Dict where key is the label and value is the colour e.g. `{'sus':'blue','vac':'red'}`
-    :param plot_type: can be 'patch' or 'line'
+    :param plot_type: can be 'patch' or 'line', or a list the same length as param_entries
     :param fig: Optionally takes in the figure to render the legend in. If not provided, a new figure will be created
+    :param legendsettings: settings for the layout of the legend. If not provided will default to appropriate values depending on whether the legend is separate or together with a plot
     :return: The matplotlib `Figure` object containing the legend
 
     """
+    if plot_type is None:
+        plot_type = 'line'
+
+    if isinstance(plot_type, list):
+        assert len(plot_type)==len(entries), 'If plot_type is a list, it must have the same number of values as there are entries in the legend (%s vs %s)'%(plot_type, entries)
+        plot_types = plot_type
+    else:
+        plot_types = [plot_type for label in entries.keys()]
 
     h = []
     for label, color in entries.items():
+        labels = entries.keys()
+        i = labels.index(label)
+        plot_type = plot_types[i]
         if plot_type == 'patch':
             h.append(Patch(color=color, label=label))
+        elif plot_type == 'line':
+            h.append(Line2D([0], [0], linewidth = settings['line_width'], color=color, label=label))
+        elif plot_type == 'circle':
+            h.append(Line2D([0], [0], marker='o', linewidth=0, markeredgewidth = settings['line_width'], fillstyle='none', color=color, label=label))
```

Comment:
In `_render_data` the scatter points are generated with `scatter()` rather than `Line2D` which means that the `line_width` setting doesn't get used - instead, the marker size and line width are hardcoded. Perhaps we should plot the legend marker the same way (using `scatter()`) and introduce settings for the marker size and edge width